### PR TITLE
fix:  Events related to different process instances may be collected in one bunch

### LINKED ISSUE #2558: Events related to different process instances may be collected in one bunch
It is possible that during execution of processes events related to different process a collected in one bunch. This leads to the fact, that wrong process instance info is stored in some events. To solve the problem _MessageProducerCommandContextCloseListener_ should be reviewed...

### PR DESCRIPTION
**Pull request by igdianov**
This PR fixes common problem that maps execution attributes for all events types as well as fixes the issue that mixes execution context attributes for multiple executions in the same transaction. 

This test ensures the execution attributes are mapped correctly for subprocesses as well as other events

Unfortunately, I had to remove execution attribute headers from message itself as there is no way to use same headers for different executions for events generated in a single transactions, so message headers will have only common runtime bundle properties...

Fixes Activiti/Activiti#2558

### CODE REVIEW

File: activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/src/main/java/org/activiti/cloud/services/events/listeners/MessageProducerCommandContextCloseListener.java [hunk 1]
```diff
@@ -54,29 +53,21 @@ public MessageProducerCommandContextCloseListener(ProcessEngineChannels producer
...
-    protected ExecutionContextInfoAppender createExecutionContextInfoAppender(ExecutionContext executionContext) {
-        return new ExecutionContextInfoAppender(executionContext);
-    }
-    
     @Override
     public void closed(CommandContext commandContext) {
         List<CloudRuntimeEvent<?, ?>> events = commandContext.getGenericAttribute(PROCESS_ENGINE_EVENTS);
         
         if (events != null && !events.isEmpty()) {
-            ExecutionContext executionContext = commandContext.getGenericAttribute(EXECUTION_CONTEXT);
```
Comment (erdemedeiros) [inline activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/src/main/java/org/activiti/cloud/services/events/listeners/MessageProducerCommandContextCloseListener.java:66]: The constant `MessageProducerCommandContextCloseListener.EXECUTION_CONTEXT` can be removed as well, as it's no longer used.

File: activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/AuditProducerIT.java [hunk 2]
```diff
@@ -556,7 +558,67 @@ public void shouldEmitEventsForTaskAddDeleteGroupCandidates() {
...
+        assertThat(subprocessIds).hasSize(2);
+        
+        String subProcessId1 = subprocessIds.get(0);
+        String subProcessId2 = subprocessIds.get(1);
+        
+        await().untilAsserted(() -> {
+            assertThat(streamHandler.getReceivedHeaders()).containsKeys(ALL_REQUIRED_HEADERS);
+
+            assertThat(streamHandler.getLatestReceivedEvents())
+                    .extracting("eventType","processInstanceId","parentProcessInstanceId", "processDefinitionKey")
```
Comment (erdemedeiros) [inline activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/AuditProducerIT.java]: Literal strings can be replaced by method references:
`.extracting(CloudRuntimeEvent::getEventType, CloudRuntimeEvent::getProcessInstanceId,CloudRuntimeEvent::getParentProcessInstanceId, CloudRuntimeEvent::getProcessDefinitionKey)`

**Review by erdemedeiros — COMMENTED** [review]
It looks good to me.

File: activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/AuditProducerIT.java (hunk 2 above)
Comment (igdianov) [reply activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/services/audit/AuditProducerIT.java]: @erdemedeiros Good suggestion. I like to use method references for extracting attributes

File: activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/src/main/java/org/activiti/cloud/services/events/listeners/MessageProducerCommandContextCloseListener.java (hunk 1 above)
Comment (igdianov) [reply activiti-cloud-services-runtime-bundle/activiti-cloud-services-events/src/main/java/org/activiti/cloud/services/events/listeners/MessageProducerCommandContextCloseListener.java:66]: @erdemedeiros Done.

**Review by salaboy — APPROVED** [review]

**Review by erdemedeiros — APPROVED** [review]

**Pull request merged**